lib/puppeteer-helpers.js

```diff
@@ -0,0 +1,56 @@
+const { componentNameToJavaScriptClassName } = require('./helper-functions.js')
+const { renderHtml } = require('./jest-helpers.js')
+
+/**
+ * Render and initialise a component within test boilerplate HTML
+ *
+ * Renders a component's Nunjucks macro with the given params, injects it into
+ * the test boilerplate page, then either:
+ *
+ * - instantiates the component class, passing the provided JavaScript
+ *   configuration, and calls the init function
+ * - runs the passed initialiser function inside the browser
+ *   (which lets you instantiate it a different way, like using `initAll`,
+ *   or run arbitrary code)
+ *
+ * @param {String} componentName - The kebab-cased name of the component
+ * @param {Object} options
+ * @param {String} options.baseUrl - The base URL of the test server
+ * @param {Object} options.nunjucksParams - Params passed to the Nunjucks macro
+ * @param {Object} [options.javascriptConfig] - The configuration hash passed to
+ *  the component's class for initialisation
+ * @param {Function} [options.initialiser] - A function that'll run in the
+ *  browser to execute arbitrary initialisation. Receives an object with the
+ *  passed configuration as `config` and the PascalCased component name as
+ *  `componentClassName`
+ * @returns {Promise<import('puppeteer').Page>}
+ */
+async function renderAndInitialise (componentName, options = {}) {
```

Comment:
Moved from [lib/jest-helpers.js](https://github.com/alphagov/govuk-frontend/pull/2878/files#diff-35cd93e336244402cc677971c753a8e36bc1ad944af90ef2b99369e1a5ba6158) due to missing `page` global

Tests vary by `@jest-environment`

```mjs
/**
 * @jest-environment puppeteer
 */
```

```mjs
/**
 * @jest-environment jsdom
 */
```